File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function sumOfLeftLeaves(root: TreeNode | null): number {
    const getLeftLeaveVal = (node: TreeNode | null, isLeft: boolean): number => {
      if (!node) {
        return 0;
      }

      if (!node.left && !node.right && isLeft) {
        return node.val;
      }

      return getLeftLeaveVal(node.left, true) + getLeftLeaveVal(node.right, false);
    };

    return getLeftLeaveVal(root, false);
};

File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        def getLeftLeaveVal(root: Optional[TreeNode], isLeft: bool) -> int:
            if not root:
                return 0

            if not root.left and not root.right and isLeft:
                return root.val

            return getLeftLeaveVal(root.left, True) + getLeftLeaveVal(root.right, False)

        return getLeftLeaveVal(root, False)